providers/running_provider.go

```diff
@@ -4,22 +4,249 @@
 package providers
 
 import (
+	"context"
+	"errors"
+	"fmt"
 	"sync"
 	"time"
 
-	"github.com/cockroachdb/errors"
 	"github.com/hashicorp/go-plugin"
 	"github.com/rs/zerolog/log"
 	pp "go.mondoo.com/cnquery/v11/providers-sdk/v1/plugin"
 	"go.mondoo.com/cnquery/v11/providers-sdk/v1/resources"
 	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
 )
 
+type connectionGraphNode struct {
+	connected bool
+	data      connectReq
+}
+
+type connectionGraph struct {
+	nodes map[uint32]connectionGraphNode
```

Comment:
could you add some more comments what we store in these properties? It's not immediately clear for me when I see it, so it may help us in case we need to change this in the future

Reply:
it would also help to describe in text how this is intended to work and put it in a comment:
- when do we add connections to the graph?
- when do we remove connections from the graph?

Reply:
Sure, will add comments here